src/State.hs

```diff
@@ -1286,6 +1286,31 @@ deleteMessage new = do
   cId <- use csCurrentChannelId
   when (postChannelId new == cId) updateViewed
 
+maybePostUsername :: ChatState -> Post -> T.Text
+maybePostUsername st p = case do { uId <- postUserId p; name <- usernameForUserId uId st; return name } of
+    Nothing -> T.empty
+    Just x -> x
+
+maybeNotify :: [PostToAdd] -> MH ()
+maybeNotify [] = do 
+    return ()
+maybeNotify (OldPost _:ps) =
+    maybeNotify ps
+maybeNotify (RecentPost p m:ps) = do
+    outputChan <- use (csResources.crSubprocessLog)
+    st <- use id
+    notifyCmd <- use (csResources.crConfiguration.to configActivityNotifyCmd)
+    case notifyCmd of
+        Nothing -> return ()
+        Just cmd ->
+            doAsyncWith Preempt $ do
+                let (messageString, notified, sender) = (T.unpack (postMessage p), if m then "1" else "2",  T.unpack (maybePostUsername st p)) 
```

Comment:
This line is a little too long. Generally we'd write this as three separate `let` bindings instead of one using a triple, i.e.,
```
let messageString = T.unpack $ postMessage p
    notified = if m then "1" else "2"
    sender = T.unpack $ maybePostUsername st p
```